### ISSUE TITLE
Color output from subprocess commands is lost

### ISSUE THREAD

**Issue by derekbrokeit** [xonsh/xonsh#1851]
This is a difference from `0.4.7` to the current `master` branch. The color output of sub process tasks are lost. This is common when piping a command, so that may be the cause of the behavior. An easy way to demonstrate this is `git status`. Typically, in most environments, this command color highlights files based on the status type. However, in the current master, the output is completely colorless.

**Comment by derekbrokeit** [comment]
Note that I am currently running my tests on a windows machine in case it is difficult to reproduce.

**Comment by melund** [comment]
This is probably related to #1842

**Comment by astronouth7303** [comment]
Are we handing commands pipes or ptys? Many commands toggle their color output based on if the file descriptor is a tty.

**Comment by scopatz** [comment]
We hand a pty if it is the last command in a pipeline and we are on posix and the command is uncaptured

**Comment by scopatz** [comment]
And a pipe otherwise

**Comment by melund** [comment]
We may get some clues from running xonsh with `readline`. Here the problem is the same. 
Some things have been fixed on master. For example the prompt, and repr output:

![image](https://cloud.githubusercontent.com/assets/1038978/19548359/5016ce44-969e-11e6-96a3-59118bbc82ce.png)

With `readline` we know that these things are also coloured with ANSI escapes. But what is the difference in the way stuff get printed in subproc output where it doesn't work? 

![image](https://cloud.githubusercontent.com/assets/1038978/19548395/89b739d6-969e-11e6-9bbd-dcc6999b22b0.png)

**Comment by laerus** [comment]
universal newlines? the ouput seems decoded, just a guess.

**Comment by melund** [comment]
If I do:
```
ls --color=auto
```
It doesn't print the color codes (But it doesn't color either). It suggests that `ls` knows it is not connected to a terminal. This is also why I can't get it to print in columns. It only does that when it thinks it is connected to a terminal.

**Comment by melund** [comment]
A workaround for `ls` is set the following alias: 

```
aliases['ls'] = ['ls', '-C', '--color=auto']
```
It will preserve the column mode even though it is thinks it is connected to a pager. And it will deactivate the colors codes. 

Are there a way we can trick the programs into thinking that they are not captured?

**Comment by scopatz** [comment]
>  Are there a way we can trick the programs into thinking that they are not captured?

Yes, this is the source of many of the remaining Windows issues.  I spent a few hours last night trying to figure this out.  POSIX uses fnctrl and ioctrl to do this, but it does not seem like there is anything in the windows API that will help.  However, I have found a lot of sack overflow pages that say powershell can do this, and some `tee con` commands claim the same. So I think there must be a way!
